### PR TITLE
Fix bugs and typos

### PR DESCRIPTION
**Pull request by jsh58**
Fix numerous bugs and typos.

### CODE REVIEW

**Comment by martenson** [comment]
Thank you John for breaking this down.

File: tools/bowtie2/bowtie2_wrapper.xml [hunk 1]
```diff
@@ -316,25 +317,25 @@
...
-                        <param name="n_ceil" type="text" value="`L,0,0.15" label="Set a function governing the maximum number of ambiguous characters (usually `N`s and/or `.`s) allowed in a read as a function of read length" help="--n-ceil; For instance, specifying `-L,0,0.15` sets the N-ceiling function `f` to `f(x) = 0 + 0.15 * x`, where x is the read length.  See also: [setting function options]. Reads exceeding this ceiling are [filtered out]. Default=`L,0,0.15`"/>
-                        <param name="dpad" type="integer" min="0" value="15" lable="Pad dynamic programming problems by that many columns on either side to allow gaps" help="--dpad; default=15"/>
+                        <param name="L" type="integer" min="0" value="20" label="Sets the length of the seed substrings to align during multiseed alignment (see `Multiseed alignment` section of help below)" help="-L; Smaller values make alignment slower but more senstive. Default=20"/>
```
Comment (nsoranzo) [inline tools/bowtie2/bowtie2_wrapper.xml]: According to Bowtie2 v2.2.4 documentation, the default value for ```-L``` in the default ```--sensitive``` mode is 22.

**Comment by nsoranzo** [comment]
Everything else looks fine, I'm merging and fixing the ```-L``` default post-merge.

**Comment by jsh58** [comment]
The default for -L is indeed 22 in --end-to-end mode, but in --local mode it is 20.  There is a similar issue with the -i parameter.  The way to fix these issues is to promote the --end-to-end / --local option to a higher level.

**Comment by martenson** [comment]
@nsoranzo Please let me know when the tool is ready to be updated in the Main TS. I will do so. Thanks.

**Comment by nsoranzo** [comment]
@jsh58 I tried to clarify in commit https://github.com/galaxyproject/tools-devteam/commit/f3db3b0ae97c5386e4c0d979a6fc9768aa0eda21#diff-5a294925ea6e37becb5eaf2ada01221eL600 , line 600-603, hope it's ok for you. I also submitted a PR for upstream bowtie2 to correct their documentation: https://github.com/BenLangmead/bowtie2/pull/17

**Comment by nsoranzo** [comment]
@martenson It's ok for me now, maybe @jsh58 would like to propose the SAM output option PR, but I think you may prefer to update in the TS anyway to fix the bugs.

**Pull request merged**